### ISSUE TITLE
Implement new Concept Exercise: `genserver`

### ISSUE THREAD

**Issue by angelikatyborska** [exercism/elixir#559]
## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)

## Goal

The student should understand the GenServer behaviour and know how to use it, writing a public module API and implementing the behaviour's callbacks.

## Learning objectives

- Know about the GenServer behaviour and its callbacks.
- Know that the name stands for "generic server".
- Understand the difference between `handle_call`, `handle_cast`, and `handle_info`.
- Understand the possible return values of the callbacks.
- Know it's a good idea to have a catch-all `handle_info` to avoid an ever-growing process inbox.
- Know that GenServers usually expose an easy-to-use public API instead of expecting its users to call `GenServer.call` themselves.
- Know when *not* to use GenServer (if `Agent` is good enough, if there is no need to keep state)

## Out of scope

- process linking, monitoring, `:DOWN` messages
- process registry, process names

## Concepts

- `genserver`

## Prerequisites

- `atoms`
- `tuples`
- `processes`
- `pid`
- `agent` (it will be easier to understand after having learned about agents)
- `use` (as in `use GenServer`)
- `behaviours`

## Practice exercises

Those practice exercises should have `genserver` in their prerequisites:
- `bank-account`
- `react`

## Resources to refer to

- https://elixir-lang.org/getting-started/mix-otp/genserver.html
- https://hexdocs.pm/elixir/master/GenServer.html

## Story

It might be a good idea to either duplicate or come up with a continuation of the stories used for the `processes` exercise (take-a-number machine) or the `agent` exercise (community garden). Ideally the story would force the student to use all three types of callbacks.

## Analyzer

The analyzer should check that indeed a module using the GenServer behaviour was implemented, and not a generic receive loop or an Agent.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
